Adds a copy action (#150)
OAI/Overlay-Specification: src/overlay.md

| Field Name | Type | Description |
| ---- | :----: | ---- |
| <a name="action-target"></a>target | `string` | **REQUIRED** A JSONPath expression selecting nodes in the target document. |

> | <a name="action-copy"></a>copy | `string` | A JSONPath expression selecting locations to copy target nodes to in the target document. If the `target` selects an object node, the value of this field MUST be an object with the properties and values to merge with the selected node. If the `target` selects an array, the value of this field MUST be an entry to append to the array. This field has no impact if the `remove` field of this action object is `true` or if the `update` field contains a value. When both the target and copy expressions match multiple nodes, the cardinality of the matches MUST be equal. Each copy node is applied to the corresponding target node in positional order: the first copy node to the first target node, the second to the second, and so forth. |

Author: (let's keep the conversation in the other thread until we come to a decision, and I'll revert back here to update the text once we have)

| ---- | :----: | ---- |
| <a name="action-target"></a>target | `string` | **REQUIRED** A JSONPath expression selecting nodes in the target document. |
| <a name="action-description"></a>description | `string` | A description of the action. [[CommonMark]] syntax MAY be used for rich text representation. |
| <a name="action-update"></a>update | Any | If the `target` selects an object node, the value of this field MUST be an object with the properties and values to merge with the selected node. If the `target` selects an array, the value of this field MUST be an entry to append to the array. This field has no impact if the `remove` field of this action object is `true` or if the `copy` field contains a value. |

> | <a name="action-copy"></a>copy | `string` | A JSONPath expression selecting locations to copy target nodes to in the target document. If the `target` selects an object node, the value of this field MUST be an object with the properties and values to merge with the selected node. If the `target` selects an array, the value of this field MUST be an entry to append to the array. This field has no impact if the `remove` field of this action object is `true` or if the `update` field contains a value. The result of the expression MUST be a object or array. |

Reviewer: ```suggestion
| <a name="action-copy"></a>copy | `string` | A JSONPath expression selecting a single node to copy into the `target` nodes. If the `target` selects an object node, the value of this field MUST be an object with the properties and values to merge with the selected node. If the `target` selects an array, the value of this field MUST be an entry to append to the array. This field has no impact if the `remove` field of this action object is `true` or if the `update` field contains a value. |
```
Remove restriction to object or array, `update` allows any type. It should be possible to copy a string from the `tags` array of one operation into the `tags` array of another operation.

The properties of the `update` object MUST be compatible with the target object referenced by the JSONPath key. When the Overlay document is applied, the properties in the `update` object are recursively merged with the properties in the target object with the same names; new properties are added to the target object.

> The properties of the resolved `copy` object MUST be compatible with the target object referenced by the JSONPath key. When the Overlay document is applied, the properties in the resolved `copy` object are recursively merged with the properties in the target object with the same names; new properties are added to the target object.

Author: I used the same linguo as for update for consistency. As far as I understand it means same object type/shape (by that I mean OpenAPI object)

> The properties of the resolved `copy` object MUST be compatible with the target object referenced by the JSONPath key. When the Overlay document is applied, the properties in the resolved `copy` object are recursively merged with the properties in the target object with the same names; new properties are added to the target object.

Author: Again, this is written in consistency with the update keyword. If we want to change that, it should be changed for both for consistency. And changing it for update most likely represents a behavior breaking change at this stage.
If we want additional keywords that impact the copy/update merge behavior, I'm not against it, but this should probably be considered outside of this scope to avoid scope creep and submitted as a separate pull request.

This approach allows inversion of control as to where the Overlay updates apply to the target document itself.

#### Copy example

> Copy actions behave similarly to update actions but source the node to from the document being transformed. Copy actions MAY be combined with update or remove actions to perform more advanced transformations like moving or renaming nodes.

Reviewer 2: Perhaps "used in sequence with" rather than "combined with" since the fields are mutually exclusive and can't be combined - so this could be confusing.

Author: updated

```yaml
openapi: 3.1.0
info:
  title: API with a paged collection
  version: 1.0.0
paths:
  /items:
    get:
      responses:
        200:
          description: OK
  /some-items:
    delete:
      responses:
        200:
          description: OK
```

```yaml
overlay: 1.1.0
info:
  title: Demonstrates variations of "copy" uses
  version: 1.0.0
actions:
  - target: '$.paths["/some-items"]'
    copy: '$.paths["/items"]'

>     description: 'copies recursively all elements from the "items" path item to the new "some-items" path item without ensuring the node exists before the copy'

Author: In case the target does not exist, same as remove/update, NO-OP
In case the copy does not exist, we could define this as NO-OP, or error? I'm happy either way.
